src/Voron/Data/BTrees/TreePage.cs

```diff
@@ -722,16 +722,17 @@ public void EnsureHasSpaceFor(LowLevelTransaction tx, Slice key, int len)
                 throw new InvalidOperationException("Could not ensure that we have enough space, this is probably a bug");
         }
 
-        public List<long> GetAllOverflowPages()
+        public long[] GetPagesReferencedBy()
         {
-            var results = new List<long>(NumberOfEntries);
+            var results = new long[NumberOfEntries];
```

Comment:
Won't this cause a LOT of allocations now? Can we avoid the allocations?

Reply:
Nope, this happens only once. And also now those arrays are not going to be bigger than 1K elements or so.

Reply:
If you ask me, I am more worried about the `MaybePrefetchPagesReferencedBy` allocations than this ones.

Reply:
Why once? This happens every time you `Seek`, which we do a lot.

Reply:
You are right. That is exactly the one I was worried about. I was under the impression you were talking about the initialization one. I was toying with the idea of using a struct iterator over a generic interface. It will avoid the allocation, completely thread safe and also generate the most efficient code when used. What do you think?

Reply:
Wouldn't a simple buffer pool be enough here?